Maybe 10

Klitsenko opened the pull request:
Endpoints for desk and employee.
Unit tests for services.

Klitsenko committed:
Merge branch 'MAYBE-6' into dev

Klitsenko committed:
Merge remote-tracking branch 'origin/dev' into dev

Klitsenko committed:
Merge remote-tracking branch 'origin/dev' into dev

Klitsenko committed:
Merge remote-tracking branch 'origin/dev' into dev

Klitsenko committed:
Merge remote-tracking branch 'origin/dev' into dev

Klitsenko committed:
Merge remote-tracking branch 'origin/dev' into dev

Klitsenko committed:
Merge remote-tracking branch 'origin/dev' into dev

Klitsenko committed:
Merge remote-tracking branch 'origin/dev' into dev

Klitsenko committed:
Endpoints for desk and employee
feature: controller\DeskController.java: basic endpoints with DML operations for desk entity;
feature: controller\EmployeeController.java: basic endpoints with DML operations for employee entity;
feature: dto\DeskRequest.java: dto for desk entity with field 'name';
feature: dto\EmployeeRequest.java: dto for employee entity with fields:
'name','login','password','roleId';
modify: entity\Desk.java: rename entity from customer
modify: entity\Employee.java: add toString() method;
modify: entity\Oder.java: rename map from 'customer' to 'desk'
modify: repository\DeskRepository.java: rename to DeskRepository.java and add method findDeskById();
modify: repository\EmployeeRepository.java: add findEmployeeById() and findEmployeeByLogin();
modify: service\DeskService.java: service for desk endpoints;
modify: service\EmployeeService.java: service for employee endpoints;
modify: resources\db\changelog\changes\change_2018.xml: rename 'customer' table to 'desk';
modify: resources\db\changelog\load\load-data.xml: rename 'customer' table to 'desk';

Klitsenko committed:
Endpoints for desk and employee
feature: controller\DeskController.java: basic endpoints with DML operations for desk entity;
feature: controller\EmployeeController.java: basic endpoints with DML operations for employee entity;
feature: dto\DeskRequest.java: dto for desk entity with field 'name';
feature: dto\EmployeeRequest.java: dto for employee entity with fields:
'name','login','password','roleId';
modify: entity\Desk.java: rename entity from customer
modify: entity\Employee.java: add toString() method;
modify: entity\Oder.java: rename map from 'customer' to 'desk'
modify: repository\DeskRepository.java: rename to DeskRepository.java and add method findDeskById();
modify: repository\EmployeeRepository.java: add findEmployeeById() and findEmployeeByLogin();
modify: service\DeskService.java: service for desk endpoints;
modify: service\EmployeeService.java: service for employee endpoints;
modify: resources\db\changelog\changes\change_2018.xml: rename 'customer' table to 'desk';
modify: resources\db\changelog\load\load-data.xml: rename 'customer' table to 'desk';

Klitsenko committed:
Refactor route
modify: controller\EmployeeController.java: change route from employee to employees;

Klitsenko committed:
Modify invoice service
modify: service\InvoiceService.java: add  employeeService.getEmployeeById() in updateFromDto();

Klitsenko committed:
Employee test
feature: test\..\service\EmployeeService.java: employee service unit test

deivur reviewed (commented):
1. Не вижу смысла от использования подобной конструкции, т.к. это все дефолтные значения.
```
return ResponseEntity.status(HttpStatus.OK)
                .contentType(MediaType.APPLICATION_JSON)
                .body(deskService.getCustomerById(id));
```
2. Не согласен с тем что название DTO обьектов заканчивается на Request. На демо Миша делал замечание, что когда читаешь код - не понятно что это DTO обьект.
    2.1 В DTO обьектах нет переопределенного toString()
3. Сервиса бизнес логики не должны возвращать налы, так это получается какой-то null driven development

YurasicV commented on src/main/java/com/maybe/maybe/controller/DeskController.java:
```diff
@@ -0,0 +1,59 @@
+    public ResponseEntity<Desk> getCustomerById(@PathVariable Long id){
+        return ResponseEntity.status(HttpStatus.OK)
+                .contentType(MediaType.APPLICATION_JSON)
+                .body(deskService.getCustomerById(id));
```
Could you try easier, like
return ResponseEntity<>(entity, status)
?

YurasicV commented on src/main/java/com/maybe/maybe/repository/EmployeeRepository.java:
```diff
@@ -4,4 +4,6 @@
 import org.springframework.data.jpa.repository.JpaRepository;
 
 public interface EmployeeRepository extends JpaRepository<Employee, Long> {
+    Employee findEmployeeById(Long id);
```
There is a default method findById
which you don't need to describe in repository class
You can use it in services

YurasicV commented on src/main/java/com/maybe/maybe/service/DeskService.java:
```diff
@@ -0,0 +1,68 @@
+        }
+    }
+
+    public List<Desk> getCustomerList() {
```
Incorrect method name

YurasicV commented on src/main/java/com/maybe/maybe/service/DeskService.java:
```diff
@@ -0,0 +1,68 @@
+        return null;
+    }
+
+    public Desk createCustomer(DeskRequest deskRequest) {
```
Incorrect method name

YurasicV commented on src/main/java/com/maybe/maybe/service/EmployeeService.java:
```diff
@@ -14,8 +20,62 @@ public EmployeeService(EmployeeRepository employeeRepository) {
-    public Employee findById(Long id) {
-        return employeeRepository.findById(id)
-                .orElseThrow(() -> new EntityNotFoundException("Could not find employee id=" + id));
+    public Employee getEmployeeById(Long id) {
```
Don't agree. To my mind, my version of this method is more correct.
Why did you recreate the method getEmployeeById?
It is clear that it returns Employee beasause it is in EmployeeService.

YurasicV commented on src/main/java/com/maybe/maybe/service/InvoiceService.java:
```diff
@@ -50,7 +50,7 @@ public Invoice createFromDTO(InvoiceDTO invoiceDTO) {
     public Invoice updateFromDTO(Invoice invoice, InvoiceDTO invoiceDTO) {
         invoice.setName(invoiceDTO.getName());
-        invoice.setEmployee(employeeService.findById(invoiceDTO.getEmployeeId()));
+        invoice.setEmployee(employeeService.getEmployeeById(invoiceDTO.getEmployeeId()));
```
Don't agree with your changes. Propose to remain findById

Klitsenko committed:
Endpoints desk and employee
refactor: controller\DeskController.java: delete media type and desk list route;
refactor: controller\EmployeeController.java:  delete media type and employee list route

Klitsenko committed:
DTO desk and employee
refactor: dto\DeskDTO.java: rename to DeskDTO and add toString();
refactor: dto\EmployeeDTO.java: rename to EmployeeDTO and add toString().

Klitsenko committed:
Entity desk
refactor: entity\Desk.java: change table name to 'desk'

Klitsenko committed:
Controller desk
refactor: controller\DeskController.java: rename method from customer to desk.

Klitsenko committed:
Abstract Entity
refactor: entity\AbstractEntity.java: delete unnecessary space

Klitsenko committed:
Desk repository
feature: repository\DeskRepository.java: add findDeskByName()

Klitsenko committed:
Desk service
refactor: service\DeskService.java: rename method from customer to desk

Klitsenko committed:
Employee service
modify: service\EmployeeService.java: delete unnecessary method

Klitsenko committed:
Data desk and employee
modify: resources\db\data\desk.csv: delete data;
modify: resources\db\data\employee.csv: delete data.

Klitsenko committed:
Desk service unit test
modify: test\..\service\DeskServiceTest.java: optimize code and add Test to the end of the test methods name

Klitsenko committed:
Employee service unit test
modify: test\..\service\EmployeeServiceTest.java: optimize code and add Test to the end of the test methods name
